FWCore/Catalog/src/FileLocator.cc

```diff
@@ -295,6 +305,16 @@ namespace edm {
     m_prefix = found_protocol->second.get("prefix", kEmptyString);
     if (m_prefix == kEmptyString) {
       //get rules
+      if (found_protocol->second.find("rules") == found_protocol->second.not_found()) {
+        fatalExceptionOccurred = true;
+        cms::Exception ex("FileCatalog");
```

Comment:
If you made a new exception which inherits from `cms::Exception` and threw that type, then in the `try..catch` block you could just catch the new exception type and rethrow it. That would do away with the odd `fatalException` boolean.

Reply:
I force pushed modifications with Chris's suggested changes. Looks better now. I'm still puzzling over the test failures. I haven't figured that out yet.